test/gitlab_merge_request_test.go

```diff
@@ -690,6 +690,135 @@ func TestGitlabMergeRequestOnUpdateAtAndLabelChange(t *testing.T) {
+	}, targetNS, forkProject.DefaultBranch,
+		triggertype.PullRequest.String(), map[string]string{})
+	assert.NilError(t, err)
+
+	targetRefName := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pac-e2e-fork-test")
+	forkCloneURL, err := scm.MakeGitCloneURL(forkProject.WebURL, opts.UserName, opts.Password)
+	assert.NilError(t, err)
+
+	commitTitle := "Add invalid .tekton file from fork - " + targetRefName
+	scmOpts := &scm.Opts{
+		GitURL:        forkCloneURL,
+		CommitTitle:   commitTitle,
+		Log:           runcnx.Clients.Log,
+		WebURL:        forkProject.WebURL,
+		TargetRefName: targetRefName,
+		BaseRefName:   originalProject.DefaultBranch,
+	}
+	_ = scm.PushFilesToRefGit(t, scmOpts, entries)
+	runcnx.Clients.Log.Infof("Pushed invalid .tekton files to fork branch: %s", targetRefName)
+
+	// Create merge request from fork to original project
+	mrTitle := "TestValidationErrorsFromFork - " + targetRefName
+	mrOptions := &clientGitlab.CreateMergeRequestOptions{
+		Title:        &mrTitle,
+		SourceBranch: &targetRefName,
+		TargetBranch: &originalProject.DefaultBranch,
+		// Create MR on the target project (original), not the source (fork)
+		TargetProjectID: &originalProject.ID,
+	}
+
+	mr, _, err := glprovider.Client().MergeRequests.CreateMergeRequest(forkProject.ID, mrOptions)
+	assert.NilError(t, err)
+	runcnx.Clients.Log.Infof("Created merge request from fork to original: %s/-/merge_requests/%d",
+		originalProject.WebURL, mr.IID)
+
+	defer func() {
+		// Close the MR and clean up branch
+		runcnx.Clients.Log.Infof("Closing MR %d", mr.IID)
+		_, _, err := glprovider.Client().MergeRequests.UpdateMergeRequest(originalProject.ID, mr.IID,
+			&clientGitlab.UpdateMergeRequestOptions{StateEvent: clientGitlab.Ptr("close")})
+		if err != nil {
+			runcnx.Clients.Log.Warnf("Failed to close MR: %v", err)
+		}
+
+		runcnx.Clients.Log.Infof("Deleting branch %s from fork", targetRefName)
+		_, err = glprovider.Client().Branches.DeleteBranch(forkProject.ID, targetRefName)
+		if err != nil {
+			runcnx.Clients.Log.Warnf("Failed to delete branch: %v", err)
+		}
+	}()
+
+	runcnx.Clients.Log.Info("Waiting for webhook validation to process MR and post validation comment...")
+
+	maxLoop := 12 // Wait up to 72 seconds for webhook processing
+	foundValidationComment := false
+
+	for i := 0; i < maxLoop; i++ {
+		notes, _, err := glprovider.Client().Notes.ListMergeRequestNotes(originalProject.ID, mr.IID, nil)
+		assert.NilError(t, err)
+
+		for _, note := range notes {
+			// Look for the validation error comment that PaC should post via webhook
+			if regexp.MustCompile(`.*There are some errors in your PipelineRun template.*`).MatchString(note.Body) &&
+				regexp.MustCompile(`.*bad-yaml\.yaml.*yaml validation error.*`).MatchString(note.Body) {
+				runcnx.Clients.Log.Info("SUCCESS: Found validation error comment on original project's MR!")
+				runcnx.Clients.Log.Info("This proves the TargetProjectID fix works in the full webhook validation flow!")
```

Comment:
Optional nit:
I'm equally excited about the fix working, but there's no reason to log if the test is successful beyond debugging😛 
```suggestion
				runcnx.Clients.Log.Debug("Found validation error comment on original project's MR")
```